### PR TITLE
Encode sim commands

### PR DESCRIPTION
**Pull request by steersbob**
- Commander sim now feeds input/output commands through an encoding/decoding cycle to check compatibility with construct.
- Removed 'object_type` from read_object command

### CODE REVIEW

File: brewblox_devcon_spark/api/object_api.py [hunk 1]
```diff
@@ -284,34 +313,34 @@ def __init__(self, app: web.Application):
...
     """
     return web.json_response(
-        await ObjectApi(request.app).all()
+        await ObjectApi(request.app).list_active()
     )
 
 
-@routes.get('/data')
-async def object_all_data(request: web.Request) -> web.Response:
+@routes.get('/saved')
```
Comment (elcojacobs) [inline brewblox_devcon_spark/api/object_api.py]: endpoints swapped for saved objects and active objects. use saved_objects and objects

File: brewblox_devcon_spark/device.py [hunk 2]
```diff
@@ -79,36 +80,42 @@ def active_profile(self, profile_id: int):
...
-            # get the type from the dict that contained data
-            obj_type = dpath.util.get(content, f'{parent}/{OBJECT_TYPE_KEY}')
-
-            # convert data, and replace in dict
-            dpath.util.set(content, f'{parent}/{OBJECT_DATA_KEY}', await processor_func(obj_type, data))
+    async def _process_data(self,
+                            processor_func: codec.TranscodeFunc_,
+                            content: dict
+                            ) -> Awaitable[dict]:
+        containers = [content]
```
Comment (elcojacobs) [inline brewblox_devcon_spark/device.py]: rename to objects_to_process? containers is a bit confusing

File: brewblox_devcon_spark/device.py [hunk 3]
```diff
@@ -146,20 +156,29 @@ def active_profile(self, profile_id: int):
 
         return service_id
 
-    async def _id_resolved(self, resolver: Callable, content: dict) -> dict:
+    async def _resolve_id(self, resolver: Callable, content: dict) -> Awaitable[dict]:
         async def resolve_key(key: str, store: datastore.DataStore):
-            for path, id in dpath.util.search(content, f'**/{key}', yielded=True):
-                dpath.util.set(content, path, await resolver(self, store, id))
+            containers = [content]
```
Comment (elcojacobs) [inline brewblox_devcon_spark/device.py]: same here

File: brewblox_devcon_spark/commands.py [hunk 4]
```diff
@@ -114,247 +98,311 @@ class Command(ABC):
...
+
+        self._check_sanity()
+
+        # `opcode` is defined as a private variable in request.
+        # it will be included in `self.encoded_request`, but not in `self.decoded_request`.
+        # `opcode` values are linked to the class, and will never deviate from the class-defined value.
+        opcode_val = self.__class__._OPCODE
+        opcode = Struct('_opcode' / Const(OpcodeEnum.encmapping[opcode_val], Byte))
+
+        # `errcode` is defined as a private variable in response.
```
Comment (elcojacobs) [inline brewblox_devcon_spark/commands.py]: comment that errors are handled as exceptions

File: brewblox_devcon_spark/api/object_api.py (hunk 1 above)
Comment (steersbob) [reply brewblox_devcon_spark/api/object_api.py]: fixed

File: brewblox_devcon_spark/device.py (hunk 2 above)
Comment (steersbob) [reply brewblox_devcon_spark/device.py]: fixed

File: brewblox_devcon_spark/device.py (hunk 3 above)
Comment (steersbob) [reply brewblox_devcon_spark/device.py]: fixed

File: brewblox_devcon_spark/commands.py (hunk 4 above)
Comment (steersbob) [reply brewblox_devcon_spark/commands.py]: fixed

**Pull request merged**